### PR TITLE
[hotfix] Improve CI Qt download stability

### LINKED ISSUE #366: Dependency download became unstable
Github Actions started to have a lot of false alarms due to download errors, e.g., https://github.com/solvcon/modmesh/actions/runs/9537805732/job/26290861151#step:5:429 :

```
WARNING : Caught ChecksumDownloadFailure, terminating installer workers
ERROR   : Failed to download checksum for the file '6.6.3-0-202403191846qtsvg-Linux-RHEL_8_8-GCC-Linux-RHEL_8_8-X86_64.7z' from mirrors '['https://download.qt.io/']
==============================Suggested follow-up:==============================
* Check your internet connection
* Consider modifying `requests.max_retries_to_retrieve_hash` in settings.ini
* Consider modifying `mirrors.trusted_mirrors` in settings.ini (see https://aqtinstall.readthedocs.io/en/stable/configuration.html#configuration)
usage: aqt [-h] [-c CONFIG]
           {install-qt,install-tool,install-doc,install-example,install-src,list-qt,list-tool,list-doc,list-example,list-src,install,tool,doc,examples,src,help,version}
           ...

Another unofficial Qt Installer.
aqt helps you install Qt SDK, tools, examples and others

options:
  -h, --help            show this help message and exit
  -c CONFIG, --config CONFIG
                        Configuration ini file.

subcommands:
  aqt accepts several subcommands:
  install-* subcommands are commands that install components
  list-* subcommands are commands that show available components
  
  commands {install|tool|src|examples|doc} are deprecated and marked for removal

  {install-qt,install-tool,install-doc,install-example,install-src,list-qt,list-tool,list-doc,list-example,list-src,install,tool,doc,examples,src,help,version}
                        Please refer to each help message by using '--help' with each subcommand
Error: Error: The process '/usr/bin/python3' failed with exit code 1
```

We need to stabilize the CI runs.  Maybe one way to improve is to cache the downloaded dependency.

### PR DESCRIPTION
**Pull request by terrychan999**
This PR fix https://github.com/solvcon/modmesh/issues/366, these are the changes:

1. Maintain the `settings.ini` of `aqtinstall` ourselves to loosen the hash algorithm for checksum from sha256 to sha1.
Also, moved `https://master.qt.io/` from mirror `fallbacks` to `trusted_mirrors`, because checksum can only be retrieved from `trusted_mirrors`. (`https://master.qt.io/` should be quite safe as the upstream)
2. Regular update `jurplel/install-qt-action` from `v3` to `v4`.
3. Enable `cache` option.

Ref:
https://aqtinstall.readthedocs.io/en/latest/configuration.html
https://github.com/jurplel/install-qt-action?tab=readme-ov-file#cache

### CODE REVIEW

File: thirdparty/aqt_settings.ini [hunk 1]
```diff
@@ -0,0 +1,249 @@
...
+min_module_size: 41
+
+[requests]
+connection_timeout: 3.5
+response_timeout: 30
+max_retries_on_connection_error: 5
+retry_backoff: 0.1
+max_retries_on_checksum_error: 5
+max_retries_to_retrieve_hash: 5
+hash_algorithm: sha1
```
Comment (terrychan999) [inline thirdparty/aqt_settings.ini:25]: here's the original
https://github.com/miurahr/aqtinstall/blob/8c4bf2ed70709b243292118cd60c19eff5480013/aqt/settings.ini#L19

File: thirdparty/aqt_settings.ini [hunk 2]
```diff
@@ -0,0 +1,249 @@
...
+INSECURE_NOT_FOR_PRODUCTION_ignore_hash: False
+
+[mirrors]
+trusted_mirrors:
+    https://download.qt.io
+    https://master.qt.io/
```
Comment (terrychan999) [inline thirdparty/aqt_settings.ini:31]: here's the original
https://github.com/miurahr/aqtinstall/blob/8c4bf2ed70709b243292118cd60c19eff5480013/aqt/settings.ini#L40

**Comment by terrychan999** [comment]
Also add a quick fix to the issue: https://github.com/solvcon/modmesh/issues/369
Not sure if I should open another PR or not.
@yungyuc @tigercosmos this PR is now ready for review, thanks.

**Review by yungyuc — APPROVED** [review]
Thanks a lot, @terrychan999 , for the quick fix.  I'd like to have it merged now, but later we would need to know more about the caching behaviors.

**Pull request merged**